src/nvim/ex_cmds.c

```diff
@@ -2116,6 +2116,16 @@ do_ecmd (
     }
   }
 
+  // Make re-editing a terminal buffer a no-op
+  if (!other_file && curbuf->terminal != NULL) {
+      // this is needed for when we are called by do_argfile() and the new
+      // argument index becomes the terminal buffer we are already editing
+      check_arg_idx(curwin);
+      maketitle();
```

Comment:
The title shows the argument list (e.g. (2 of 5)), because we can have switched to a new entry (when this function was called from ```do_argfile()```), we need to update the title.